### ISSUE TITLE
4.0.0 beta: Javascript tests don't execute on successive runs

### ISSUE THREAD

**Issue by smoove** [trufflesuite/truffle#585]
- [x] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue

When running tests from inside the new "develop" console, the javascript tests don't get executed on successive runs. The solidity tests work as expected.

## Steps to Reproduce

```
$ truffle init
$ truffle develop
truffle(develop)> test
```

*Notice that both "TestMetacoin" and "Contract: MetaCoin" tests get executed*

Run tests again:

```
truffle(develop)> test
```

*Notice that only "TestMetacoin" test gets executed*

## Environment

* Operating System: Ubuntu 16.04
* Truffle version: 4.0.0 beta
* Ethereum client: truffle develop
* node version: latest stable
* npm version: latest stable

**Comment by tcoulter** [comment]
Thanks for the bug report. Looks like a regression (we've had this issue before). Will take a look!

**Comment by gnidan** [comment]
This has been fixed incidentally as part of https://github.com/trufflesuite/truffle-core/pull/47 and will be part of the next beta release.

**Comment by amexperts** [comment]
I just upgraded to:

Truffle v4.0.0-beta.2 (core: 4.0.0-beta.2)
Solidity v0.4.17 (solc-js)

and am getting the same issue as above. I am running on Linux within Virtualbox running on Windows 10 Professional.

**Comment by 3esmit** [comment]
I'm also having this issue.

**Comment by gnidan** [comment]
This should be fixed in Truffle v4.0.0. Closing!
